### ISSUE TITLE
Array property item insert / remove not reflected in store via events

### ISSUE THREAD

**Issue by Palivonas** [feathers-plus/feathers-vuex#189]
### Steps to reproduce

Patch a model that has an array property (e.g. `someModel.nestedItems`)
Any removals / additions to the array property are not reflected in other clients even though real-time events arrive as they should.

A repo to reproduce: https://github.com/Palivonas/feathers-vuex-bug-reproduction
* `npm install` backend and frontend, run `npm start` and `npm run serve`.
* Open localhost:8080, add an item (`create` called immediately)
* Open a second browser window, click "disable cloning" in that second window, this will use item reference directly to render the form (instead of  a clone) in order to see live changes
* Try adding or removing items in the first window (and clicking save)

### Expected behavior
Items are added/removed to/from item.nestedItems throughout all connected clients

### Actual behavior
Items are not added/removed in the second client instance. Data is correct after a refresh.

No such issue with FeathersVuex version 1.1.4 (using store action instead of model instances, cause it's not available in 1.1.4)
Events are arriving via websockets without problems.

**Module versions**:
```
"@feathersjs/feathers": "^3.3.0",
"@feathersjs/socketio-client": "^1.2.0",
"feathers-vuex": "^1.6.2",
"socket.io-client": "^2.2.0",
```

**NodeJS version**: 10

**Operating System**: MacOS

**Browser Version**: Chrome 71

**Module Loader**: Webpack (Vue CLI default setup)

**Comment by metareason** [comment]
@Palivonas we are encountering similar issues with arrays in models not taking updates correctly, try setting `replaceItems: true` in your service definition in your store, this will cause the mutation to fully replace the Model instead of merging.

Still looks like there is a bug in the way _merge() works at https://github.com/feathers-plus/feathers-vuex/blob/d2a240adeef67173c06d41b9d435907a464ab695/src/service-module/mutations.js#L56 - needs further investigation.

**Comment by Palivonas** [comment]
@metareason I did try `replaceItems: true` (should've mentioned in the issue I guess), but it did not help, unfortunately.

I'll try to find which release/commit introduced this bug (a chance to finally play with `git-bisect`!). Hopefully will have some time for it this week.

**Comment by dkebler** [comment]
Related maybe I have a property that holds an array.  If I dispatch a patch with a new array for that property new elements of array are patched (what?), but if I remove elements from the current array the patch does not remove them.  Weird cause I kinda assumed that patch would simply replace the entire array with the new one but apparently it walks the existing array merging new elements but not removing items not there.  Totally not expected.   

Maybe _merge is walking the enumerable values of the property value as well.  Why I don't know but it is apparently doing that and merging the elements of the array instead replacing the complete array with the patched one.

**Comment by marshallswain** [comment]
This is fixed in `feathers-vuex@1.7.0-pre.44`.  `npm i feathers-vuex@pre`.  You can see the implementation and test in the above-referenced commit.  Thanks for pointing this out, everybody.  Thanks for the issue @Palivonas 🍻
